### PR TITLE
Backport palette packing asm from dav1d 1.3.0

### PR DESCRIPTION
**Pull request by fbossen**
Move palette packing/edge-extension into a DSP function and provide asm and Rust equivalents

### CODE REVIEW

**Comment by fbossen** [comment]
This is on top of #987

**Comment by kkysen** [comment]
> This is on top of #987

You can set the target branch to the branch for #987.  You might have to move the branches to `memorysafety/rav1d` instead of `fbossen/rav1d`, though, not sure.

**Comment by fbossen** [comment]
> You can set the target branch to the branch for #987. You might have to move the branches to `memorysafety/rav1d` instead of `fbossen/rav1d`, though, not sure.

Thank you for the suggestion. Indeed, I think I would have to move the branches to `memorysafety/rav1d` for this to work

File: src/pal.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            }
+        }
+        src = &src[bw..];
+        if y < h - 1 {
+            dst = &mut dst[dst_bw..];
+        }
+    }
+
+    if h < bh {
+        let (last_row, dst) = dst.split_at_mut(dst_bw);
```
[the remark this answers is not in the document]
Comment (fbossen) [reply src/pal.rs:61]: Yes. I add `dst_bw` one fewer times in the previous loop, so I don't have to deal with a negative index here.

**Review by kkysen — APPROVED** [review]
LGTM, but I'm not sure why it's causing some timeouts on aarch64 macOS, since these changes shouldn't make things slower.

**Pull request merged**